xray/src/event_loop.rs

```diff
@@ -190,19 +191,18 @@ impl EventLoop {
                     .as_micros();
                 self.packets[send_index].recv_index = Some(self.recv_counter as u64);
                 self.packets[send_index].recv_ts = Some(recv_ts);
-                self.recv_counter += 1;
                 self.on_maybe_recv_all(finish_timeout);
             }
         }
         Ok(())
     }
 
     fn on_maybe_recv_all(&self, finish_timeout: &mut Pin<&mut tokio::time::Sleep>) {
-        if self.recv_counter > self.cli_args.packet_count {
-            println!("All packets were received. Waiting 3 seconds to make sure pcap is properly populated");
+        if self.recv_counter >= self.cli_args.packet_count {
+            println!("All packets were received. Finishing test...");
             finish_timeout
                 .as_mut()
-                .reset(Instant::now() + Duration::from_secs(3));
+                .reset(Instant::now() + Duration::from_secs(1));
```

Comment:
This might not be necessary at all going forward, at least it's not necessary at the moment, but it might be necessary in a separate PR so reducing the timeout is better for now than removing the timeout completely